### DISCUSSION TITLE
XSD generate

### DISCUSSION THREAD

**Discussion by aovzerk** [vezel-dev/novadrop#64] [Questions] [answered]
How do you generate xsd files?

**Comment by alexrp** [comment, marked as answer]
99% of XSDs are generated by walking the data tree and empirically observing the schema that seems to have been used.

The other 1% can only be done by human intervention; in general, there is no way to distinguish a truly recursive schema from one that only appears to be recursive.

A few examples:

* `/TaskDef/TaskClass/Param/Param/...` is a case of real recursion.
* `/WorldData/MoneyString/MoneyString/...` is not actually recursive.
* `/SkillData/Skill/Property/Property/...` is recursive.
* `/FootStepSound/FootStepSound/...` is not.
* etc...

**Reply by aovzerk** [reply]
what program did you use to create xsd, can you share?

**Reply by alexrp** [reply]
There's not really any canonical program for that. @Roukanken42 hacked some messy code together to do it for the patch 115 data center when we were working on the first release of Novadrop. We then fixed up any recursion issues manually.

You'd probably be better off writing your own program to do it if you want.

**Reply by aovzerk** [reply]
Thanks for the answer
